app/utilities/protocol_importers/protocols_io/v3/protocol_normalizer.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module ProtocolImporters
+  module ProtocolsIO
+    module V3
+      class ProtocolNormalizer < ProtocolImporters::ProtocolNormalizer
+        def load_protocol(id)
+          response = ProtocolImporters::ProtocolsIO::V3::ApiClient.new.single_protocol(id).parsed_response
+
+          normalized_hash = normalize response[:protocol]
+          normalized_hash
+        end
+
+        private
+
+        def normalize(protocol_hash)
+          normalized_data = {}
+          normalized_data[:source_id] = protocol_hash[:id]
+          normalized_data[:name] = protocol_hash[:title]
+          normalized_data[:description] = santz(protocol_hash[:description])
+          normalized_data[:authors] = protocol_hash[:authors].map { |e| e[:name] }.join(', ')
+
+          normalized_data[:steps] = protocol_hash[:steps].map do |e|
+            {
+              source_id: e[:id],
+              name: e[:components].find { |o| o[:type_id] == 6 }[:source][:title],
```

Comment:
Step component digging will be used a lot. Should probably be refactored into a special function.

Additionally it would be better to use descriptive function names for every component that we should use (e.g. `step_name(e[:components])`. Else with this magic numbering the code will be something to look at in a couple of months ([hint hint](https://github.com/biosistemika/scinote-web/blob/7812f70c2f04624a093a5247e4cf13c332c58cf8/app/views/protocols/import_export/_import_json_protocol_s_desc.html.erb)). You can go even a step further and define a mapping of all magic numbers to respective constants or function, which can then be used:

```
def get_component(id, components):
    if AVAILABLE_COMPONENTS.include?(id):
      components.find { |o| o[:type_id] == id}
    else
      raise ArgumentError
    end
end
       
def step_name(components)
    get_component(6, components)[:source][:title]
end
```

A utility module for handling components would be awesome. Saying this here, since it's going to get messier and messier with every step we take. (pun almost intended)